src/com/sungardas/init/CredentialsServiceImpl.java

```
@@ -152,27 +157,34 @@ public boolean checkDefaultUser(String login, String password) {
             for (Bucket bucket : allBuckets) {
                 try {
-                    ListObjectsRequest request = new ListObjectsRequest()
-                            .withBucketName(bucket.getName()).withPrefix("sdfsstate");
-                    if (client.listObjects(request).getObjectSummaries().size() > 0) {
-                        if (bucketName.equals(bucket.getName())) {
-                            result.get(0).setCreated(true);
-                        } else {
-                            String location = client.getBucketLocation(bucket.getName());
-
-                            if (!location.equalsIgnoreCase(currentLocation))
-                                continue;
-                            result.add(new InitConfigurationDto.S3(bucket.getName(), true));
+                    if (bucket.getName().startsWith(ENHANCED_SNAPSHOT_BUCKET_PREFIX)) {
+                        String location = client.getBucketLocation(bucket.getName());
+
+                        // Because client.getBucketLocation(bucket.getName()) returns US if bucket is in us-east-1
+                        if (!location.equalsIgnoreCase(currentLocation) && !location.equalsIgnoreCase("US")) {
+                            continue;
+                        }
+
+                        ListObjectsRequest request = new ListObjectsRequest()
+                                .withBucketName(bucket.getName()).withPrefix("sdfsstate");
+                        if (client.listObjects(request).getObjectSummaries().size() > 0) {
+                            if (bucketName.equals(bucket.getName())) {
+                                result.get(0).setCreated(true);
+                            } else {
+                                result.add(new InitConfigurationDto.S3(bucket.getName(), true));
+                            }
                         }
                     }
-                } catch (AmazonS3Exception ignored) {
+                } catch (Exception ignored) {

```

- For general exceptions like this add comment in the code to explain why the exception is ignored
- Ensure the detail about the exception is logged in